app/src/main/java/com/ifttt/groceryexpress/LocationForegroundService.kt

```diff
@@ -0,0 +1,71 @@
+package com.ifttt.groceryexpress
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.Service
+import android.content.Context
+import android.content.Intent
+import android.graphics.Color
+import android.os.Build.VERSION.SDK_INT
+import android.os.Build.VERSION_CODES.O
+import android.os.IBinder
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+
+/**
+ * Example implementation of a foreground service: a persistent service constantly running on the background with a
+ * visible notification.
+ *
+ * The purpose of this service is to make sure the Grocery Express app can get more frequent location updates, without
+ * the background location limit.
+ *
+ * More details see https://developer.android.com/about/versions/oreo/background-location-limits.
+ */
+class LocationForegroundService : Service() {
+    override fun onBind(intent: Intent?): IBinder? = null
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        val notificationManager = NotificationManagerCompat.from(this)
+
+        val notification = NotificationCompat.Builder(this, NOTIFICATION_CHANNEL_ID).setSmallIcon(R.mipmap.ic_launcher)
+            .setContentTitle(getString(R.string.foreground_notification_title))
+            .setColor(Color.BLACK)
+            .build()
+
+        if (SDK_INT >= O) {
```

Comment:
Is this a typo? I believe we want to check if SDK_INT >= 8?

Reply:
This is `android.os.Build.VERSION_CODES.O`.

Reply:
ah